Dockerfile

```diff
@@ -0,0 +1,31 @@
+FROM golang:1.14-alpine AS build
+
+ARG LDFLAGS
+
+# Install some dependencies needed to build the project
+RUN apk add bash ca-certificates git gcc g++ libc-dev
+
+# All these steps will be cached
+RUN mkdir /build
+WORKDIR /build
+
+# Copying the .mod and .sum files before the rest of the code
+# to improve the caching behavior of Docker
+COPY go.mod . 
+COPY go.sum .  
+
+# Get dependancies - will also be cached if we won't change mod/sum
+RUN go mod download
+# COPY the source code as the last step
+COPY . .
+
+# And compile the project 
+# CGO_ENABLED and installsuffix are part of the scheme to get better caching on builds
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -ldflags="${LDFLAGS}" -a -installsuffix cgo -o /go/bin/zqd ./cmd/zqd
+
+
+FROM golang:1.14-alpine
```

Comment:
I don't think zqd needs anything from this image, so maybe `FROM scratch`?

Reply:
Hi Noah -- I super-appreciate your detailed review of the Dockerfile, and I tried to get "FROM scratch" working, because it would reduce the image size by 150MB. Let me know if you have any suggestion for this cert problem:

When zqd runs, it accesses S3, so it needs to have up-to-date ca-certs. Otherwise, zqd give the error:

> RequestError: send request failed
> caused by: Get "https://brim-scratch.s3.us-east-2.amazonaws.com/?delimiter=%2F&list-type=2&prefix=mark%2Fzqd-meta%2F": x509: certificate signed by unknown authority
> 

The problem with "FROM scratch" is that I do not know how to install the certs in that case. For alpine, this works:

RUN apk update && apk add ca-certificates

But "FROM scratch" gives an error because there is no package manager to install the certs:

> => ERROR [stage-1 2/3] RUN apk update && apk add ca-certificates                                                                                                                0.2s
> ------
>  > [stage-1 2/3] RUN apk update && apk add ca-certificates:
> #5 0.224 container_linux.go:349: starting container process caused "exec: \"/bin/sh\": stat /bin/sh: no such file or directory"
> ------
> failed to solve with frontend dockerfile.v0: failed to build LLB: executor failed running [/bin/sh -c apk update && apk add ca-certificates]: runc did not terminate sucessfully

Reply:
Thanks, Noah! This was a great suggestion. In order to get 'FROM scratch' working, I had to put 'apk add ca-certificates' back into the build container, and then I could copy just the certs from build to scratch with:

> COPY --from=build /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/ca-certificates.crt

I am super happy because now we have a 22.8 MB docker image for zqd!

Reply:
Nice.